File: app/src/main/java/guepardoapps/mycoins/activities/ActivityBoot.java
package guepardoapps.mycoins.activities;

import android.app.Activity;
import android.os.Bundle;
import android.os.Handler;

import guepardoapps.mycoins.R;
import guepardoapps.mycoins.controller.NavigationController;
import guepardoapps.mycoins.controller.SettingsController;

public class ActivityBoot extends Activity {
    private static final String TAG = ActivityBoot.class.getSimpleName();
    private Logger _logger;

    private NavigationController _navigationController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.side_boot);

        _logger = new Logger(TAG, true);
        _logger.Debug("onCreate");

        _navigationController = new NavigationController(this);

        SettingsController.getInstance().Initialize(this);
        SettingsController.getInstance().Dispose();
    }

    @Override
    protected void onResume() {
        super.onResume();
        _logger.Debug("onResume");
        navigateToMain();
    }

    @Override
    protected void onPause() {
        super.onPause();
        _logger.Debug("onPause");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        _logger.Debug("onDestroy");
    }

    private void navigateToMain() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                _navigationController.NavigateTo(MainActivity.class, true);
            }
        }, 1500);
    }
}

File: app/src/main/java/guepardoapps/mycoins/activities/ActivityBoot.kt
package guepardoapps.mycoins.activities

import android.app.Activity
import android.os.Bundle
import android.os.Handler
import com.github.guepardoapps.timext.kotlin.extensions.seconds
import com.github.guepardoapps.timext.kotlin.postDelayed
import guepardoapps.mycoins.R
import guepardoapps.mycoins.common.Constants
import guepardoapps.mycoins.controller.NavigationController
import guepardoapps.mycoins.controller.SharedPreferenceController
import guepardoapps.mycoins.services.coin.CoinService
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers

class ActivityBoot : Activity() {

    private var subscription: Disposable? = null

    @ExperimentalUnsignedTypes
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.side_boot)

        val sharedPreferenceController = SharedPreferenceController(this)

        if (!sharedPreferenceController.load(Constants.sharedPrefName, false)) {
            sharedPreferenceController.run {
                save(Constants.currency, Constants.currencyDefault)
                save(Constants.reloadEnabled, Constants.reloadEnabledDefault)
                save(Constants.reloadTimeoutMs, Constants.reloadTimeoutMsDefault)
                save(Constants.sharedPrefName, true)
            }
        }

        if (!CoinService.instance.isInitialized) {
            subscription = CoinService.instance.initialSetupPublishSubject
                    .subscribeOn(Schedulers.io())
                    .subscribe(
                            { done ->
                                if (done) {
                                    NavigationController(this).navigate(ActivityMain::class.java, true)
                                }
                            },
                            { }
                    )
        } else {
            Handler().postDelayed({ NavigationController(this).navigate(ActivityMain::class.java, true) }, 1.5.seconds)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        subscription?.dispose()
    }
}